src/engine_render.c

```diff
@@ -6027,73 +6211,197 @@ static void update_frontview_pointed_block(unsigned long laaa, unsigned char qdr
     {
-    case 1:
-        coord_y -= slb_width;
-        coord_z += slb_width;
+    //case 0: // North
+    case 1: // East
+        coord_y -= box_height;
+        coord_z += box_height;
         break;
-    case 2:
-        coord_x -= slb_width;
-        coord_y -= slb_width;
-        coord_z += slb_width;
+    case 2: // South
+        coord_x -= box_width;
+        coord_y -= box_height;
+        coord_z += box_height;
         break;
-    case 3:
-        coord_x -= slb_width;
+    case 3: // West
+        coord_x -= box_width;
         break;
     }
-    vstart = 0;
     coord_z -= (stl_width >> 1);
-    vend = stl_width;
-    delta[0] = 0;
-    delta[1] = slb_width;
-    delta[2] = depth;
-    delta[3] = slb_width + depth;
-    // Draw a horizonal line element for every subtile
-    for (i=3; i > 0; i--)
-    {
-      if (!is_free_space_in_poly_pool(4))
+    // Draw 4 horizonal line elements
+    create_line_element(coord_x,             coord_y,                      coord_x + box_width, coord_y,                      coord_z,                          map_volume_box.color);
+    create_line_element(coord_x,             coord_y + box_height,         coord_x + box_width, coord_y + box_height,         coord_z - box_height,             map_volume_box.color);
+    create_line_element(coord_x,             coord_y + depth,              coord_x + box_width, coord_y + depth,              coord_z,                          map_volume_box.color);
+    create_line_element(coord_x,             coord_y + box_height + depth, coord_x + box_width, coord_y + box_height + depth, coord_z - box_height,             map_volume_box.color);
+    // Now the lines at left and right
+    create_line_element(coord_x,             coord_y,                      coord_x,             coord_y + box_height,         coord_z - box_height,             map_volume_box.color);
+    create_line_element(coord_x + box_width, coord_y,                      coord_x + box_width, coord_y + box_height,         coord_z - box_height,             map_volume_box.color);
+    create_line_element(coord_x,             coord_y + breadth,            coord_x,             coord_y + box_height + depth, coord_z - box_height + stl_width, map_volume_box.color);
+    create_line_element(coord_x + box_width, coord_y + breadth,            coord_x + box_width, coord_y + box_height + depth, coord_z - box_height + stl_width, map_volume_box.color);
+}
+void create_accurate_frontview_map_volume_box(struct RoomMap room_map, struct Camera *cam, unsigned char stl_width)
+{
+    unsigned char orient = ((unsigned int)(cam->orient_a + LbFPMath_PI/4) >> 9) & 0x03;
+    int floor_height_z = (map_volume_box.floor_height_z == 0) ? 1 : map_volume_box.floor_height_z; // ignore "liquid height", and force it to "floor height". All fancy rooms are on the ground, and this ensures the boundboxes are drawn correctly. A different solutino will be required if this function is used to draw fancy rooms over "liquid".
```

Comment:
solutino is not a word

Reply:
I think it is the Italian spelling ;p (Thank you I've corrected it)